### PR TITLE
Fix regexp matching bidi control characters

### PR DESCRIPTION
**Pull request by arlolra**
/cc @wikimedia/services @subbuss

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -345,9 +345,9 @@ Title.newFromText = function(title, siteInfo, defaultNs) {
 
     title = title
     // Strip Unicode bidi override characters.
-    .replace(/\xE2\x80[\x8E\x8F\xAA-\xAE]/g, '')
+    .replace(/[\u200E\u200F\u202A-\u202E]/g, '')
     // Clean up whitespace
-    .replace(/[ \xA0\u1680\u180E\u2000-\u200A\u2028\u2029\u202F\u205F\u3000]+/g, '_')
+    .replace(/[ _\u00A0\u1680\u180E\u2000-\u200A\u2028\u2029\u202F\u205F\u3000]+/g, '_')
```
Comment (Pchelolo) [inline lib/index.js]: Minor - why do we need to replace `_` back to `_`?

Comment (arlolra) [inline lib/index.js]: Because of the + sign at the end of this regexp.  See the test I added `[ 'en.wikipedia.org', 'Foo _ bar', 'Foo_bar' ],`

Comment (Pchelolo) [inline lib/index.js]: Oh, right! So many corner cases..
